panimg/image_builders/dicom.py

```diff
@@ -267,21 +276,52 @@ def _create_itk_from_dcm(self, z_order: int) -> SimpleITK.Image:
         photometric_interpretation = getattr(
             self.ref_header, "PhotometricInterpretation", None
         )
-        if not is_rgb and photometric_interpretation == "MONOCHROME1":
-            if np.issubdtype(dcm_array.dtype, np.floating):
-                dcm_array = -dcm_array
-            else:
-                dcm_array = ~dcm_array
+        self._pixel_values_need_inverting = (
+            not is_rgb and photometric_interpretation == "MONOCHROME1"
+        )
+        dcm_array = self._invert_pixel_values_if_needed(dcm_array)
 
         return SimpleITK.GetImageFromArray(dcm_array, isVector=is_rgb)
 
     def _add_optional_metadata(self, img: SimpleITK.Image):
         for f in OPTIONAL_METADATA_FIELDS:
             value = getattr(self.ref_header, f, "")
             str_value = str(value)
-            if str_value != "":
-                validate_metadata_value(key=f, value=value)
-                img.SetMetaData(f, str_value)
+
+            # Skip empty entries, ITK will not write them anyway
+            if str_value == "":
+                continue
+
+            validate_metadata_value(key=f, value=value)
+            img.SetMetaData(f, str_value)
+
+    def _add_window_level_metadata(self, img: SimpleITK.Image):
+        try:
+            centers = self.ref_header.WindowCenter
+            widths = self.ref_header.WindowWidth
+        except AttributeError:
+            return
+
+        validate_metadata_value(key="WindowCenter", value=centers)
+        validate_metadata_value(key="WindowWidth", value=widths)
+
+        centers = np.asarray(centers, dtype=np.float32)
+        widths = np.asarray(widths, dtype=np.float32)
```

Comment:
I would let numpy handle finding the right data type. For these values it defaults either to an int64 or float64 but it would prevent any (unexpected) decimal precision.